tests/tests_lite/conftest.py

```diff
@@ -83,6 +85,46 @@ def reset_deterministic_algorithm():
     torch.use_deterministic_algorithms(False)
 
 
+def mock_cuda_count(monkeypatch, n: int) -> None:
+    monkeypatch.setattr(lightning_lite.accelerators.cuda, "num_cuda_devices", lambda: n)
+    monkeypatch.setattr(lightning_lite.accelerators.cuda, "num_cuda_devices", lambda: n)
+
+
+@pytest.fixture(scope="function")
+def cuda_count_0(monkeypatch):
+    mock_cuda_count(monkeypatch, 0)
+
+
+@pytest.fixture(scope="function")
+def cuda_count_1(monkeypatch):
+    mock_cuda_count(monkeypatch, 1)
+
+
+@pytest.fixture(scope="function")
+def cuda_count_2(monkeypatch):
+    mock_cuda_count(monkeypatch, 2)
+
+
+@pytest.fixture(scope="function")
+def cuda_count_4(monkeypatch):
+    mock_cuda_count(monkeypatch, 4)
```

Comment:
@awaelchli I initially didn't add these to the Lite conftest because mocking it is just one line in for Lite. Also the tests are still using

`@mock.patch("lightning_lite.accelerators.cuda.num_cuda_devices", return_value=0)`

So we could remove them from this PR